### PR TITLE
Fixed identified dependency analysis issues.

### PR DESCRIPTION
**Pull request by cobratbq**
Fixed issues reported in `dependency:analyze` analysis report regarding undeclared dependencies and unused dependencies.

### CODE REVIEW

File: pom.xml [hunk 1]
```diff
@@ -281,25 +309,43 @@
...
-                <groupId>org.simplify4u</groupId>
-                <artifactId>slf4j-mock</artifactId>
-                <version>2.1.0</version>
-            </dependency>
+        <dependency>
+            <groupId>org.simplify4u</groupId>
+            <artifactId>slf4j-mock</artifactId>
+            <version>2.1.0</version>
+            <scope>test</scope>
+        </dependency>
```
Comment (slawekjaranowski) [inline pom.xml:340]: pleas fix indent

File: pom.xml [hunk 2]
```diff
@@ -467,6 +567,16 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
+                <version>3.1.2</version>
```
Comment (slawekjaranowski) [inline pom.xml:573]: version is defined in parent pom

File: pom.xml [hunk 3]
```diff
@@ -467,6 +567,16 @@
                     </execution>
                 </executions>
             </plugin>
+            <plugin>
+                <groupId>org.apache.maven.plugins</groupId>
+                <artifactId>maven-dependency-plugin</artifactId>
```
Comment (slawekjaranowski) [inline pom.xml:572]: we can add goal `analyze-only` to standard build

File: pom.xml [hunk 4]
```diff
@@ -467,6 +567,16 @@
...
+                <version>3.1.2</version>
+                <configuration>
```
Comment (slawekjaranowski) [inline pom.xml:574]: we should skip execution of plugin when tests are skiped, so if add goal `analyze-only`
please also add
```xml
<skip>${maven.test.skip}</skip>
```

**Review by slawekjaranowski — CHANGES_REQUESTED** [review]

File: pom.xml (hunk 3 above)
Comment (cobratbq) [reply pom.xml:572]: In that case, I will also add `failOnWarning` to ensure that findings are apparent.

**Review by slawekjaranowski — APPROVED** [review]

**Pull request merged**